mealie/repos/repository_recipes.py

```diff
@@ -152,57 +196,41 @@ def page_all(
             joinedload(RecipeModel.tools),
         ]
 
-        item_class: type[RecipeSummary | RecipeSummaryWithIngredients]
-
-        if load_food:
-            args.append(joinedload(RecipeModel.recipe_ingredient).options(joinedload(RecipeIngredient.food)))
-            args.append(joinedload(RecipeModel.recipe_ingredient).options(joinedload(RecipeIngredient.unit)))
-            item_class = RecipeSummaryWithIngredients
-        else:
-            item_class = RecipeSummary
-
         q = q.options(*args)
 
         fltr = self._filter_builder()
         q = q.filter_by(**fltr)
 
         if cookbook:
-            cb_filters = self._category_tag_filters(
-                cookbook.categories,
-                cookbook.tags,
-                cookbook.tools,
-                cookbook.require_all_categories,
-                cookbook.require_all_tags,
-                cookbook.require_all_tools,
+            cb_filters = self._build_recipe_filter(
+                categories=extract_uuids(cookbook.categories),
+                tags=extract_uuids(cookbook.tags),
+                tools=extract_uuids(cookbook.tools),
+                require_all_categories=cookbook.require_all_categories,
+                require_all_tags=cookbook.require_all_tags,
+                require_all_tools=cookbook.require_all_tools,
             )
 
             q = q.filter(*cb_filters)
+        else:
```

Comment:
I assumed that there is really no case where both cookbook and the other filters will be set at the same time, because it really doesnt make much sense,  but please let me know if this is a bad idea for any reason